tools/buildsys/src/builder.rs

```diff
@@ -156,10 +150,6 @@ impl PackageBuildArgs {
         args.build_arg("VARIANT_FLAVOR", &self.variant_flavor);
         args.build_arg("VARIANT_PLATFORM", &self.variant_platform);
         args.build_arg("VARIANT_RUNTIME", &self.variant_runtime);
-        for image_feature in &self.image_features {
-            args.build_arg(format!("{}", image_feature), "1");
-        }
```

Comment:
We need to finish the conditional compilation removal in `os.spec` before it's safe to do this.

The image features serve two roles:
- Outward-facing, to tell cargo when to rebuild.
- Inward-facing, where they control the bconds we generate so the spec file can include or exclude certain programs.

If we stop passing in these arguments then the bconds generated in the "rpm-macros-and-bconds" stage of the Dockerfile will be incorrect for package builds.

We really need to clean up https://github.com/bottlerocket-os/bottlerocket/pull/3828 and get it merged before we can land this change.

Note that the first commit in that PR removes the bconds, but we can't take that in isolation because pluto still can't be built for a non-k8s model until the following patch.

Reply:
@bcressey check the latest push. All functionality is retained but gated behind the `BUILDSYS_DEPRECATED_FEATURE` flag.